### PR TITLE
[autoscaler] Add unit tests for stopped node caching, fix flaky tests

### PR DESCRIPTION
**Pull request by ericl**

### CODE REVIEW

**Comment by richardliaw** [comment]
Can you assign this to someone else? Thanks.

File: python/ray/autoscaler/autoscaler.py [hunk 1]
```diff
@@ -324,7 +325,7 @@ def _launch_node(self, config, count):
                 TAG_RAY_NODE_NAME: "ray-{}-worker".format(
                     config["cluster_name"]),
                 TAG_RAY_NODE_TYPE: "worker",
```
Comment (edoakes) [inline python/ray/autoscaler/autoscaler.py]: Make "worker" and "head" constants too while you're at it?

File: python/ray/autoscaler/autoscaler.py [hunk 2]
```diff
@@ -666,7 +659,9 @@ def should_update(self, node_id):
         if not self.can_update(node_id):
             return (None, None, None)
 
-        if self.files_up_to_date(node_id):
+        status = self.provider.node_tags(node_id).get(TAG_RAY_NODE_STATUS)
+
```
Comment (edoakes) [inline python/ray/autoscaler/autoscaler.py]: remove space

File: python/ray/autoscaler/autoscaler.py [hunk 3]
```diff
@@ -666,7 +659,9 @@ def should_update(self, node_id):
         if not self.can_update(node_id):
```
Comment (edoakes) [inline python/ray/autoscaler/autoscaler.py]: Should have a comment here that (None, None, None) means should not update

File: python/ray/autoscaler/autoscaler.py [hunk 4]
```diff
@@ -666,7 +659,9 @@ def should_update(self, node_id):
...
             return (None, None, None)
```
Comment (edoakes) [inline python/ray/autoscaler/autoscaler.py]: Don't need parens here and below

**Review by edoakes — APPROVED** [review]
LGTM just some small comments

**Pull request merged**